src/NServiceBus.Core/UnitOfWork/TransactionScopes/TransactionScopeUnitOfWorkBehavior.cs

```diff
@@ -16,8 +16,7 @@ public HandlerTransactionScopeWrapperBehavior(TransactionOptions transactionOpti
         {
             if (Transaction.Current != null)
             {
-                await next().ConfigureAwait(false);
-                return;
+                throw new Exception("Ambient transaction already present, please remove the transaction scope unit of work by removing the call to .UnitOfWork().WrapHandlersInATransactionScope()");
```

Comment:
We already prevent that here:

https://github.com/Particular/NServiceBus/pull/3249/files#diff-6640594deaddf558586f25d67587be32R14

So the only way for this to happen is if the users creates a behavior with a scope him self?